pkgs/tools/graphics/gfxreconstruct/default.nix

```diff
@@ -0,0 +1,83 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+, cmake
+, makeWrapper
+, pkg-config
+, python3
+, wayland
+, libX11
+, libxcb
+, lz4
+, vulkan-loader
+, xcbutilkeysyms
+, zlib
+, zstd
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gfxreconstruct";
+  version = "0.9.15";
+
+  src = fetchFromGitHub {
+    owner = "LunarG";
+    repo = "gfxreconstruct";
+    rev = "v${version}";
+    hash = "sha256-hIzQ5L0Payj8hlyy5UI7RXgnyhQBPqG7nfbvW2VYvTg=";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [
+    libX11
+    libxcb
+    lz4
+    python3
+    wayland
+    xcbutilkeysyms
+    zlib
+    zstd
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    makeWrapper
+    pkg-config
+  ];
+
+  # The python script searches in subfolders, but we want to search in the same bin directory
+  prePatch = ''
+    substituteInPlace tools/gfxrecon/gfxrecon.py \
+      --replace "scriptdir, '..', cmd" 'scriptdir'
+
+    # Fix build with gcc11
```

Comment:
67-52 = 15 lines, all of them affecting C++ source code, none of them related to Nix/NixOS/Nixpkgs quirks.

Why don't you

1. create a `.patch` file?
2. send the patch to upstream developers too?

Reply:
Looks like someone already did that. It compiles without these now.